browser-extension/mv3/src/service-worker/services/scriptRuleHandler.ts

```diff
@@ -16,14 +16,21 @@ export const applyScriptRules = async (tabId: number, frameId: number, url: stri
   const appliedScriptRuleIds = new Set<string>();
 
   scriptRules.forEach((scriptRule) => {
-    scriptRule.pairs.forEach((scriptRulePair: ScriptRulePair) => {
-      if (matchSourceUrl(scriptRulePair.source, url)) {
+    if (
+      matchRuleWithRequest(scriptRule, {
+        url,
+        type: frameId === 0 ? ResourceType.MainDocument : ResourceType.IFrameDocument,
+        method: "GET",
+        initiator: url,
```

Comment:
Incase of iframe, Url is the iframe url here. It should be the main frame url instead